components/application-gateway/internal/proxy/unauthorizedresponseretrier.go

```diff
@@ -53,15 +55,14 @@ func (rr *retrier) retry() (*http.Response, error) {
 	request, cancel := rr.prepareRequest()
 	defer cancel()
 
-	var err error
+	request.Body = rr.body
```

Comment:
I suggest adding a test case corresponding to the issue that we are solving so it will not occur again in case of some future changes. The best place for that would be probably `proxy_test.go`

Reply:
Done!